src/libstd/sys/windows/process.rs

```diff
@@ -393,7 +393,11 @@ impl From<c::DWORD> for ExitStatus {
 
 impl fmt::Display for ExitStatus {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "exit code: {}", self.0)
+        if self.0 & 0x80000000 != 0 {
```

Comment:
It would be nice to have a comment explaining this heuristic briefly.

Reply:
@varkor Apparently, `ExitProcess` is a newtype for `u32` on Windows. And Windows uses sometimes [system error codes](https://docs.microsoft.com/en-gb/windows/desktop/Debug/system-error-codes), sometimes [NTSTATUS](https://docs.microsoft.com/en-us/windows-hardware/drivers/kernel/using-ntstatus-values).

The `self.0 & 0x80000000 != 0` for `u32` acts like `self.0 < 0` for `i32`, but I'd render informational status codes (the ones within `0x40000000 − 0x7FFFFFFF` range) as hexadecimal as well.

Reply:
I'm not sure why only *large* codes are printed in hexadecimal. What's the problem with printing all codes in hexadecimal? We wouldn't need any heuristics and comments then.
(Small codes are documented and searchable in both hexadecimal and decimal forms.)

Reply:
Are there any examples of exceptions crashing the process that have an informational status codes though?

Reply:
@petrochenkov Because system error codes are decimal. For example, `12000 - 12175` is reserved for [Internet Error codes](https://docs.microsoft.com/en-gb/windows/desktop/Debug/system-error-codes--12000-15999-). So, displaying it in hexadecimal would be the opposite of this PR.